modules/lambda-registrator/main.tf

```diff
@@ -127,8 +156,51 @@ resource "aws_iam_role_policy_attachment" "lambda_logs" {
   policy_arn = aws_iam_policy.policy.arn
 }
 
+resource "aws_ecr_repository" "lambda-registrator" {
+  count        = var.enable_pull_through_cache ? 0 : 1
+  name         = var.private_ecr_repo_name
+  force_delete = true
+}
+
+
+resource "aws_ecr_pull_through_cache_rule" "pull_through_cache_rule" {
+  count                 = var.enable_pull_through_cache ? 1 : 0
+  ecr_repository_prefix = var.ecr_repository_prefix
+  upstream_registry_url = var.upstream_registry_url
+}
+
+resource "docker_image" "lambda_registrator" {
+  name = var.enable_pull_through_cache ? local.ecr_image_uri_pull_through : var.consul_lambda_registrator_image
+  depends_on = [
+    aws_ecr_pull_through_cache_rule.pull_through_cache_rule
+  ]
+}
+
+resource "docker_tag" "lambda_registrator_tag" {
+  count        = var.enable_pull_through_cache ? 0 : 1
+  source_image = docker_image.lambda_registrator.name
+  target_image = local.ecr_image_uri
+}
+
+resource "null_resource" "push_image" {
```

Comment:
We originally didn't include this because it breaks TFC. It looks like this PR only allows using a pull through cache or pushing docker images. This effectively requires TFC users to use the pull through cache approach.

Is this a tradeoff we are ok with?